riscv-pages/src/page.rs

```diff
@@ -39,6 +39,16 @@ impl PageSize {
         (val + PageSize::Size4k as u64 - 1) / (PageSize::Size4k as u64)
     }
 
+    /// Return the number of pages of this `PageSize` that will fit in `val`
```

Comment:
I am slightly unclear on either the description of purpose of this function. My guess is that the values for the page size match the number of bits (sans the PFN) or something along those lines.

Reply:
It's the same thing as  val / PageSize

Doing with shifts is faster and you don't have to think about what happens with the remainder.  It's less confusing than trying to round to the nearest page, etc. Just shift.

it could also be written

```rust
match self {
    PageSize::Size4k => val >> 12;
    PageSize::Size2M => val >>  19,
    PageSize::Size1G => val >> 30,
    PageSize::Size512G => val >> 39,
}
```

Reply:
It's fine the way as written, but wasn't sure of the exact intent. I think I understand it now, but will read the code a little more.